Compilation of samples/drivers/flash_shell still fails on arduino_nano_33_iot:

Here the relevant output:
```
INFO    - 500/[577](https://github.com/zephyrproject-rtos/zephyr/runs/5630973355?check_suite_focus=true#step:11:577) arduino_nano_33_iot       samples/drivers/flash_shell/sample.drivers.flash.shell FAILED Build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/arduino_nano_33_iot/samples/drivers/flash_shell/sample.drivers.flash.shell/build.log
ERROR   - Loading Zephyr default modules (Zephyr base).
```
... and then ...
```
[135/166] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_sam0.c.obj
FAILED: zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_sam0.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/arm/atmel_sam0/samd21 -I../../../../../../lib/util/fnmatch/. -I../../../../../../drivers -I../../../../../../soc/arm/atmel_sam0/common/. -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/hal/atmel/asf/sam0/include/samd21 -isystem ../../../../../../lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include -isystem /opt/toolchains/zephyr-sdk-0.13.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include-fixed -Os -imacros /__w/zephyr/zephyr/twister-out/arduino_nano_33_iot/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0plus -mthumb -mabi=aapcs -mfp16-format=ieee -imacros /__w/zephyr/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/drivers/flash_shell=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_sam0.c.obj -MF zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_sam0.c.obj.d -o zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_sam0.c.obj -c /__w/zephyr/zephyr/drivers/flash/flash_sam0.c
In file included from ../../../../../../soc/arm/atmel_sam0/samd21/soc.h:34,
                 from /__w/zephyr/zephyr/drivers/flash/flash_sam0.c:17:
/__w/zephyr/zephyr/drivers/flash/flash_sam0.c: In function 'flash_sam0_write_protection':
/__w/zephyr/modules/hal/atmel/asf/sam0/include/samd21/samd21g18a.h:535:31: error: implicit declaration of function '_UL' [-Werror=implicit-function-declaration]
  535 | #define FLASH_SIZE            _UL(0x00040000) /* 256 kB */
      |                               ^~~
/__w/zephyr/zephyr/drivers/flash/flash_sam0.c:36:27: note: in expansion of macro 'FLASH_SIZE'
   36 | #define LOCK_REGION_SIZE (FLASH_SIZE / LOCK_REGIONS)
      |                           ^~~~~~~~~~
/__w/zephyr/zephyr/drivers/flash/flash_sam0.c:394:17: note: in expansion of macro 'LOCK_REGION_SIZE'
  394 |       offset += LOCK_REGION_SIZE) {
      |                 ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```
The macro`_UL` which is used by the HAL module in `/zephyrproject/modules/hal/atmel/asf/sam0/include/samd21/samd21g18a.h` seems not to be defined anywhere accessible...

Should I create an issue for this?

--- a/boards/arm/arduino_nano_33_iot/arduino_nano_33_iot.dts
+++ b/boards/arm/arduino_nano_33_iot/arduino_nano_33_iot.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 #include <atmel/samd21.dtsi>
 #include "arduino_nano_33_iot-pinctrl.dtsi"
+#include "arduino_nano_r3_connector.dtsi"
 
 / {
 	model = "Arduino Nano 33 IOT";

--- a/boards/arm/arduino_nano_33_iot/arduino_nano_r3_connector.dtsi
+++ b/boards/arm/arduino_nano_33_iot/arduino_nano_r3_connector.dtsi
@@ -1,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Joylab AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	arduino_nano_header: connector {
+		compatible = "arduino-nano-header-r3";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <0 0 &portb 23 0>,  /* D0 / UART-RX */
+			   <1 0 &portb 22 0>,  /* D1 / UART-TX */
+			   <2 0 &portb 10 0>,  /* D2 */
+			   <3 0 &portb 11 0>,  /* D3 */
+			   <4 0 &porta 7 0>,   /* D4 */
+			   <5 0 &porta 5 0>,   /* D5 */
+			   <6 0 &porta 4 0>,   /* D6 */
+			   <7 0 &porta 6 0>,   /* D7 */
+			   <8 0 &porta 18 0>,  /* D8 */
+			   <9 0 &porta 20 0>,  /* D9 */
+			   <10 0 &porta 21 0>, /* D10 */
+			   <11 0 &porta 16 0>, /* D11 / SPI-MOSI */
+			   <12 0 &porta 19 0>, /* D12 / SPI-MISO */
+			   <13 0 &porta 17 0>, /* D13 / SPI-SCK */
+			   <14 0 &porta 2 0>,  /* D14 / A0 */
+			   <15 0 &portb 2 0>,  /* D15 / A1 */
+			   <16 0 &porta 11 0>, /* D16 / A2 */
+			   <17 0 &porta 10 0>, /* D17 / A3 */
+			   <18 0 &portb 8 0>,  /* D18 / A4 / I2C-SDA */
+			   <19 0 &portb 9 0>,  /* D19 / A5 / I2C-SCL */
+			   <20 0 &porta 9 0>,  /* D20 / A6 */
+			   <21 0 &portb 3 0>;  /* D21 / A7 */
+	};
+};
+
+arduino_nano_i2c: &sercom4 {};
+arduino_nano_spi: &sercom1 {};
+arduino_nano_serial: &sercom5 {};

--- a/boards/arm/nucleo_g031k8/arduino_nano_r3_connector.dtsi
+++ b/boards/arm/nucleo_g031k8/arduino_nano_r3_connector.dtsi
@@ -1,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Joylab AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	arduino_nano_header: connector {
+		compatible = "arduino-nano-header-r3";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <0 0 &gpiob 7 0>,   /* D0 / UART-RX */
+			   <1 0 &gpiob 6 0>,   /* D1 / UART-TX */
+			   <2 0 &gpioa 15 0>,  /* D2 */
+			   <3 0 &gpiob 1 0>,   /* D3 */
+			   <4 0 &gpioa 10 0>,  /* D4 */
+			   <5 0 &gpioa 9 0>,   /* D5 */
+			   <6 0 &gpiob 0 0>,   /* D6 */
+			   <7 0 &gpiob 2 0>,   /* D7 */
+			   <8 0 &gpiob 8 0>,   /* D8 */
+			   <9 0 &gpioa 8 0>,   /* D9 */
+			   <10 0 &gpiob 9 0>,  /* D10 */
+			   <11 0 &gpiob 5 0>,  /* D11 / SPI-MOSI */
+			   <12 0 &gpiob 4 0>,  /* D12 / SPI-MISO */
+			   <13 0 &gpiob 3 0>,  /* D13 / SPI-SCK */
+			   <14 0 &gpioa 0 0>,  /* D14 / A0 */
+			   <15 0 &gpioa 1 0>,  /* D15 / A1 */
+			   <16 0 &gpioa 4 0>,  /* D16 / A2 */
+			   <17 0 &gpioa 5 0>,  /* D17 / A3 */
+			   <18 0 &gpioa 12 0>, /* D18 / A4 / I2C-SDA */
+			   <19 0 &gpioa 11 0>, /* D19 / A5 / I2C-SCL */
+			   <20 0 &gpioa 6 0>,  /* D20 / A6 */
+			   <21 0 &gpioa 7 0>;  /* D21 / A7 */
+	};
+};
+
+arduino_nano_i2c: &i2c2 {};
+arduino_nano_spi: &spi1 {};
+arduino_nano_serial: &usart1 {};

--- a/boards/arm/nucleo_g031k8/nucleo_g031k8.dts
+++ b/boards/arm/nucleo_g031k8/nucleo_g031k8.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 #include <st/g0/stm32g031X8.dtsi>
 #include <st/g0/stm32g031k(4-6-8)tx-pinctrl.dtsi>
+#include "arduino_nano_r3_connector.dtsi"
 
 / {
 	model = "STMicroelectronics STM32G031K8-NUCLEO board";

--- a/dts/bindings/gpio/arduino-nano-header-r3.yaml
+++ b/dts/bindings/gpio/arduino-nano-header-r3.yaml
@@ -1,0 +1,38 @@
+# Copyright (c) 2022 Joylab AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    GPIO pins exposed on Arduino Nano (R3) headers.
+
+    The Arduino Nano layout provides two headers on opposite edges of the board.
+
+    * A 15-pin header with mostly digital signals. The additional NRST (pin3)
+      and GND (pin 4) pins are not exposed by this binding.
+    * A 15-pin Analog Input and power supply header. This has analog input
+      signals labeled from A0 through A7, as well a digital signal D13. The
+      power supply and reset pins are not exposed by this binding.
+
+    This binding provides a nexus mapping for 22 pins where parent pins 0
+    through 13 correspond to D0 through D13, and parent pins 14 through 21
+    correspond to A0 through A7, as depicted below.
+
+        1  D1                   VIN    -
+        0  D0                   GND    -
+        -  RESET                RESET  -
+        -  GND                  5V     -
+        2  D2                   A7/D21 21
+        3  D3                   A6/D20 20
+        4  D4                   A5/D19 19
+        5  D5                   A4/D18 18
+        6  D6                   A3/D17 17
+        7  D7                   A2/D16 16
+        8  D8                   A1/D15 15
+        9  D9                   A0/D14 14
+        10 D10                  AREF   -
+        11 D11                  3V3    -
+        12 D12                  D13    13
+
+
+compatible: "arduino-nano-header-r3"
+
+include: [gpio-nexus.yaml, base.yaml]

